lib/CL/devices/almaif/AlmaifDB/AlmaifDBManager.cc

```diff
@@ -0,0 +1,289 @@
+/* AlmaifDBManager.cc - Parses and and responds to queries about AlmaifDB
+
+   Copyright (c) 2023 Topi Leppänen / Tampere University
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+
+#include "AlmaifDBManager.hh"
+
+#include "../AlmaifShared.hh"
+
+#include "pocl_file_util.h"
+
+#include <dirent.h>
+#include <iostream>
+#include <set>
+
+#include "tiny-json.h"
+
+void AlmaifDBManager::parseOverlay(json_t const *overlay,
```

Comment:
```suggestion
void AlmaIFDatabaseManager::parseOverlay(json_t const *overlay,
```
AlmaIFBitstreamDatabase would sound more descriptive. DB can store anything.